bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/services/UrlProviderImpl.java

```diff
@@ -201,14 +202,19 @@ private static Resource toSpecificPage(Resource page, Set<String> selectors, Sli
                 continue;
             }
 
+            // get the filterType property set by the picker
+            String filterType = jcrContent.getValueMap().get(SELECTOR_FILTER_TYPE_PROPERTY, "uidAndUrlPath");
+
             // The property is saved as a String when it's a simple selection, or an array when a multi-selection is done
             String[] selectorFilters = filter.getClass().isArray() ? ((String[]) filter) : ArrayUtils.toArray((String) filter);
 
             // When used with the category picker and the 'uidAndUrlPath' option, the values might have a format like 'Mjg=|men/men-tops'
             // --> so we split them to first extract the category ids
+            // V2 of the component uses 'urlPath' and does not requiere any processing
             Set<String> selectorFiltersSet = Arrays.asList(selectorFilters)
                 .stream()
-                .map(s -> (StringUtils.contains(s, UID_AND_URL_PATH_SEPARATOR) ? StringUtils.substringAfter(s, UID_AND_URL_PATH_SEPARATOR)
+                .map(s -> ((StringUtils.equals(filterType, "uidAndUrlPath") && StringUtils.contains(s, UID_AND_URL_PATH_SEPARATOR))
```

Comment:
Instead of doing the comparison inside the `map()` you can do it outside avoiding the unnecessary mapping for the else case.

trivial: prefer method references over lambdas
trivial: you may want to use `Arrays#stream()` directly instead of `Arrays#asList()`.

```java
Stream<String> selectorFilterStream = Arrays.stream(selectorFilters);
if (StringUtils.equals(filterType, "uidAndUrlPath")) {
 selectorFilterStream = selectorFilterStream
   .map(s -> StringUtils.contains(s, UID_AND_URL_PATH_SEPARATOR) 
    ? StringUtils.substringAfter(s, UID_AND_URL_PATH_SEPARATOR)
    : s);
}
Set<String> selectorFilterSet = selectorFilterStream
 .filter(StringUtils::isNotEmpty)
 .collect(Collectors.toSet());
```

Reply:
Updated that part